app/api/session/session.tsx

```diff
@@ -1,6 +1,32 @@
+import "server-only";
+import { JWTPayload, SignJWT, jwtVerify } from "jose";
 import { NextResponse } from "next/server";
-export async function createSession({response, exp, session_id}:{response: NextResponse, exp: number, session_id: string}) {
-    response.cookies.set("session_id", session_id, {
+
+const sessionSecretKey = process.env.SESSION_SECRET;
+const encodedKey = new TextEncoder().encode(sessionSecretKey);
+
+export async function encrypt({ envelope, exp }:{ envelope: JWTPayload, exp: number }) {
+    return new SignJWT(envelope)
+        .setProtectedHeader({ alg: "HS256" })
+        .setIssuedAt()
+        .setExpirationTime(exp)
+        .sign(encodedKey)
+}
```

Comment:
Having worked and played w/ this now, I would ask if we could add some way of error-ing out earlier than when we try to sign. Even right before would give us the control to throw a more sensible error for the next developer.

Reply:
Added an error